attestation-service/verifier/src/se/ibmse.rs

```diff
@@ -0,0 +1,318 @@
+        let encr_measurement_key =
+            encrypt_measurement_key(conf_data.measurement_key(), rsa_public_key);
+        let encr_request_nonce =
+            encrypt_measurement_key(conf_data.nonce().clone().unwrap().value(), rsa_public_key);
+
+        Ok(Self {
+            request_blob,
+            measurement_size: meas_alg.exp_size(),
+            additional_size: arcb.flags().expected_additional_size(),
+            encr_measurement_key,
+            encr_request_nonce,
+            image_hdr_tags: *image_hdr_tags,
+        })
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SeAttestationResponse {
+    #[serde_as(as = "Base64")]
+    measurement: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    additional_data: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    user_data: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    cuid: ConfigUid,
+    #[serde_as(as = "Base64")]
+    encr_measurement_key: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    encr_request_nonce: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    image_hdr_tags: BootHdrTags,
+}
+
+impl SeAttestationResponse {
+    pub fn from_slice(response: &[u8]) -> Result<Self> {
+        Ok(serde_json::from_slice(response).unwrap())
+    }
+
+    pub fn from_string(request: &str) -> Result<Self> {
+        Ok(serde_json::from_str(request).unwrap())
+    }
+
+    pub fn create(
+        measurement: &[u8],
+        additional_data: &[u8],
+        user_data: &[u8],
+        cuid: &ConfigUid,
+        encr_measurement_key: &[u8],
+        encr_request_nonce: &[u8],
+        image_hdr_tags: &BootHdrTags,
+    ) -> Result<Self> {
+        Ok(Self {
+            measurement: measurement.to_vec(),
+            additional_data: additional_data.to_vec(),
+            user_data: user_data.to_vec(),
+            cuid: *cuid,
+            encr_measurement_key: encr_measurement_key.to_vec(),
+            encr_request_nonce: encr_request_nonce.to_vec(),
+            image_hdr_tags: *image_hdr_tags,
+        })
+    }
+
+    pub fn verify(&self, priv_key_file: &str) -> Result<SeAttestationClaims> {
+        let mut file = File::open(priv_key_file)?;
+        let mut contents = vec![];
+        file.read_to_end(&mut contents)?;
+
+        let rsa = Rsa::private_key_from_pem(&contents)?;
+        let rsa_private_key = &PKey::from_rsa(rsa)?;
+
+        let meas_key = decrypt_measurement_key(&self.encr_measurement_key, rsa_private_key);
+        let nonce = decrypt_measurement_key(&self.encr_request_nonce, rsa_private_key);
+
+        if nonce.len() != 16 {
+            return Err(anyhow!("The nonce vector must have exactly 16 elements."));
+        }
+        let boxed_slice: Box<[u8]> = nonce.into_boxed_slice();
+        let boxed_array: Box<[u8; 16]> = match boxed_slice.try_into() {
+            Ok(ba) => ba,
+            Err(_) => return Err(anyhow!("Failed to convert nonce from Vec<u8> to [u8; 16].")),
+        };
+        let nonce_array: [u8; 16] = *boxed_array;
```

Comment:
decrypt_measurement_key returns a Vec<u8>, so we can do just that, no?

```rust
let nonce_array: [u8; 16] = nonce
  .try_into()
  .ok_or(anyhow!("Failed to convert nonce from Vec<u8> to [u8; 16]."))?;
```

Reply:
Fixed, thanks!